### DISCUSSION TITLE
Luigi with SAP Fiori Launchpad (FLP)

### DISCUSSION THREAD

**Discussion by madhusree4594** [SAP/luigi#3213] [Q&A]
Hi Team,

I am building micro frontend project structure using Luigi framework.
My Launch page built using FLP Module and also I have loaded Luigi scripts in the same.

I am getting following error, 
<img width="1565" alt="image" src="https://user-images.githubusercontent.com/53636255/224276369-5a78e723-cc2a-465d-bdff-b47a34fd79a5.png">

Can someone help me on this

**Comment by hardl** [comment]
Hi, looks like Luigi router and FLP router are interfering.
If you want to integrate Luigi core in an app that has similar capabilities, you have to make sure to resolve these kind of conflicts.
For example wrt routing, you could define `skipRoutingForUrlPatterns` (https://docs.luigi-project.io/docs/navigation-parameters-reference/?section=skiproutingforurlpatterns) for routes that should be ignored by luigi. Not sure, if FLP has sth. similar.
In general, I would recommend to wait for the release of Luigi Container (there is a RC already) for integration use cases in existing apps. More information can be found here: https://github.com/SAP/luigi/tree/main/container.

**Reply by madhusree4594** [reply]
Hi,

Thanks for the reply. I tried setting skipRoutingForUrlPatterns for my initial routing. But, I am getting this following message when I am trying to navigate from Main application to other application 
<img width="1559" alt="image" src="https://user-images.githubusercontent.com/53636255/225513207-0ff45fe3-6f95-48bf-ab1a-ff52d4256fac.png">

**Reply by hardl** [reply]
Hi, that usually happens, if you execute LuigiClient commands too early, before the handshake btw core app and micro frontend is completed.
Make sure your luigi-related logic is initiated from the initListener https://docs.luigi-project.io/docs/luigi-client-api/?section=addinitlistener.

**Reply by madhusree4594** [reply]
Hi, Thanks for the suggestion, I will try this
